### DISCUSSION TITLE
How to use shared enums?

### DISCUSSION THREAD

**Discussion by fleeser** [serverpod/serverpod#714] [Q&A]
I created a project named project_shared and created enums in there that I would like to use inside the typical Serverpod yaml files, but unfortunately when running `serverpod generate` inside the generated files the enum is underlined red and it says 

> The name 'FriendRequestStatus' isn't a type, so it can't be used as a type argument.
> Try correcting the name to an existing type, or defining a type named 'FriendRequestStatus'

How do I fix this or is there no support for this type of approach. The thing here is that I need those enum values in other classes inside project_shared. That's why I put them there.

**Comment by yahu1031** [comment]
import the shared package into the server

**Reply by fleeser** [reply]
It is imported.

**Comment by fischerscode** [comment]
I'd recommend letting Serverpod generate the enum, just like classes. (This is currently not documented.) https://github.com/serverpod/serverpod/blob/eec8af12e41aef94cc9f162358d9bb3e77bd9eef/tests/serverpod_test_server/lib/src/protocol/test_enum.yaml#L1-L12

**Reply by fleeser** [reply]
I think you forgot that I wrote custom serializable classes inside the project_shared project that also need the enum values.

**Comment by fleeser** [comment]
Isn't there something like the following for enums?

```
extraClasses:
  - package:my_shared_package/my_shared_package.dart:FreezedCustomClass
```
